cypress/integration/unit_tests/polygon_draw_test.js

```diff
@@ -49,142 +67,182 @@ describe('Unit test for ShapeData', () => {
         {'SNAP HOUSEHOLDS': 1000, 'TOTAL HOUSEHOLDS': 1000});
     const scoreCollection =
         ee.FeatureCollection([feature1, feature2, feature3]);
-    // Polygon contains only first damage point.
+    // Default polygon contains only first damage point.
     damageCollection = ee.FeatureCollection([
       ee.Feature(ee.Geometry.Point([1.5, 1.5])),
       ee.Feature(ee.Geometry.Point([200, 200])),
     ]);
     // Use our custom EarthEngine FeatureCollections.
     cy.stub(resourceGetter, 'getScoreAsset').returns(scoreCollection);
 
-    // Set up appropriate Firestore mocks.
-    cy.stub(getFirestoreRoot(), 'collection')
-        .withArgs('usershapes')
-        .returns(firebaseCollection);
-    for (const prop in firebaseCollection) {
-      if (firebaseCollection.hasOwnProperty(prop)) {
-        delete firebaseCollection[prop];
-      }
-    }
-    // Make sure .get() succeeds.
-    firebaseCollection.get = () => [];
-    // Make sure userShapes is set in the code, and use our custom EarthEngine
-    // FeatureCollections.
-    return cy.wrap(initializeAndProcessUserRegions(null, Promise.resolve({
-      data: () => ({asset_data: {damage_asset_path: damageCollection}}),
-    })));
+    return setUpPage();
   });
 
-  it('Add shape', () => {
-    const popup = new StubPopup();
-    const underTest = new StoredShapeData(null, null, null, popup);
-    const records = [];
-    firebaseCollection.add = recordRecord(records, {id: 'new_id'});
-    popup.notes = 'my notes';
-    cy.wrap(underTest.update()).then(() => {
-      expect(records).to.eql([{
-        calculatedData: calculatedData,
-        geometry: polygonGeometry,
-        notes: 'my notes',
-      }]);
-      expect(underTest.id).to.eql('new_id');
-      expect(StoredShapeData.pendingWriteCount).to.eql(0);
-    });
+  /**
+   * Initializes the Google Map; sets up the polygon drawing manager; and sets
+   * up user feature handling. No features are expected on first call, since the
+   * test Firestore database is empty.
+   * @return {Cypress.Chainable}
+   */
+  function setUpPage() {
+    return createGoogleMap()
+        .then((mapResult) => map = mapResult)
+        .then(() => setUpPolygonDrawing(map, Promise.resolve(null)))
+        .then((drawingManagerResult) => drawingManager = drawingManagerResult)
+        .then(() => {
+          userRegionData.clear();
+          return initializeAndProcessUserRegions(map, Promise.resolve({
+            data: () => ({asset_data: {damage_asset_path: damageCollection}}),
```

Comment:
this is setting the damage_asset_path to the actual feature collection but in real life it's a string/asset path that resolves to a feature collection. correct?